### PR TITLE
Make sendTransaction broadcast the signed transaction

### LINKED ISSUE #283: `sendTransaction` API must broadcast the signed transacion
this way we allow an end to end API workflow for developers, but also we help mitigating #271 problem, giving clear visibility via explorer and returning the transaction hash instead.

this issue must change:

- `marina-provider` type definition, changing the result type
- `docs.vulpem.com`
- implement in the backend

### PR DESCRIPTION
**Pull request by bordalix**
fixes #283

sendTransaction no longer return the signed tx hex, now broadcast it and return the transaction id.

@tiero please review

### CODE REVIEW

File: src/content/marinaBroker.ts [hunk 1]
```diff
@@ -180,7 +181,10 @@ export default class MarinaBroker extends Broker {
 
           if (!accepted) throw new Error('the user rejected the create tx request');
           if (!signedTxHex) throw new Error('something went wrong with the tx crafting');
-          return successMsg(signedTxHex);
+
+          const txid = await broadcastTx(selectEsploraURL(state), signedTxHex);
+          if (!txid) throw new Error('something went wrong with the tx broadcasting');
```
Comment (tiero) [inline src/content/marinaBroker.ts:186]: would be wroth to have here a console.debug printing the signedTxHex maybe?

Comment (tiero) [reply src/content/marinaBroker.ts:186]: also would be useful to catch the Esplora error coming out of it

Comment (bordalix) [reply src/content/marinaBroker.ts:195]: Done.

Comment (tiero) [reply src/content/marinaBroker.ts:195]: missing console.debug of the signedTxHex

Comment (bordalix) [reply src/content/marinaBroker.ts:195]: Is there: https://github.com/bordalix/marina/blob/broadcast_tx_283/src/content/marinaBroker.ts#L185

**Comment by tiero** [comment]
bumped marina-provider to 1.4.5, please update deps

**Pull request merged**